### ISSUE TITLE
Mainline build failing since #24275 on Ubuntu 24.04

### ISSUE THREAD

**Issue by xnox** [openssl/openssl#24383]
```console
docker run -ti --rm ubuntu:noble
sed 's/deb/deb deb-src/' -i /etc/apt/sources.list.d/ubuntu.sources
apt update
apt build-dep openssl
apt install git
git clone https://github.com/openssl/openssl
cd openssl
./Configure
make -j$(nproc)

# make test TESTS=test_sysdefault
"make" depend && "make" _tests
make[1]: Entering directory '/openssl'
make[1]: Leaving directory '/openssl'
make[1]: Entering directory '/openssl'
"make" run_tests
make[2]: Entering directory '/openssl'
( SRCTOP=. \
  BLDTOP=. \
  PERL="/usr/bin/perl" \
  FIPSKEY="f4556650ac31d35461610bac4ed81b1a181b2d8a43ea2854cbae22ca74560813" \
  EXE_EXT= \
  /usr/bin/perl ./test/run_tests.pl test_sysdefault )
00-prep_fipsmodule_cnf.t .. skipped: FIPS module config file only supported in a fips build
Files=1, Tests=0,  0 wallclock secs ( 0.00 usr  0.00 sys +  0.08 cusr  0.00 csys =  0.08 CPU)
Result: NOTESTS
90-test_sysdefault.t .. 1/? 
    # ERROR: (ptr) 'ctx = SSL_CTX_new(TLS_method()) != NULL' failed @ test/sysdefaulttest.c:25
    # 0x0
    # 800B38523A760000:error:0A000180:SSL routines:SSL_CONF_cmd:bad value:ssl/ssl_conf.c:946:cmd=SignatureAlgorithms, value=RSA+SHA256:nonex
    # 800B38523A760000:error:0A0001A3:SSL routines:SSL_CTX_new_ex:error in system default config:ssl/ssl_lib.c:4100:
    # OPENSSL_TEST_RAND_SEED=1715591272
    not ok 1 - test_func
90-test_sysdefault.t .. ok   ---------------------------------------------------
All tests successful.
Files=1, Tests=3,  0 wallclock secs ( 0.01 usr  0.01 sys +  0.10 cusr  0.05 csys =  0.17 CPU)
Result: PASS
make[2]: Leaving directory '/openssl'
make[1]: Leaving directory '/openssl'
```

f6ce48f5b8ad4d8d748ea87d2490cbed08db9936 is last passing commit
21819f78b057c254254646a7854bfad0cd40ed83 is first failing commit which is from https://github.com/openssl/openssl/pull/24275

**Comment by mattcaswell** [comment]
...and yet, in spite of the "ERROR" output you seem to get:

````
All tests successful.
Files=1, Tests=3,  0 wallclock secs ( 0.01 usr  0.01 sys +  0.10 cusr  0.05 csys =  0.17 CPU)
Result: PASS
````

**Comment by mattcaswell** [comment]
I'm seeing the same behaviour with Ubuntu 23.10

**Comment by xnox** [comment]
Yeah, I don't know if this test case is expected-failure or not; and if "not ok" is correct or if errors are correct or if summary is correct. It is all weird.

**Comment by mattcaswell** [comment]
Fix in #24384

**Comment by mattcaswell** [comment]
> Yeah, I don't know if this test case is expected-failure or not; and if "not ok" is correct or if errors are correct or if summary is correct. It is all weird.

The test is a negative test to ensure that a bad config file results in errors being raised. Unfortunately those errors are being displayed as error output running the test.
